checks.d/mysql.py

```diff
@@ -32,78 +33,341 @@
+    'Innodb_buffer_pool_bytes_free': ('mysql.innodb.buffer_pool_free', GAUGE),
+    'Innodb_buffer_pool_bytes_used': ('mysql.innodb.buffer_pool_used', GAUGE),
+    'Innodb_buffer_pool_bytes_total': ('mysql.innodb.buffer_pool_total', GAUGE),
+    'Innodb_buffer_pool_read_requests': ('mysql.innodb.buffer_pool_read_requests', RATE),
+    'Innodb_buffer_pool_reads': ('mysql.innodb.buffer_pool_reads', RATE),
+    'Innodb_buffer_pool_pages_utilization': ('mysql.innodb.buffer_pool_utilization', GAUGE),
 }
 
 
+# Calculated from "SHOW MASTER LOGS;"
+BINLOG_VARS = {
+    'Binlog_space_usage_bytes': ('mysql.binlog.disk_use', GAUGE),
+}
+
+# Additional Vars found in "SHOW STATUS;"
+# Will collect if [FLAG NAME] is True
+OPTIONAL_STATUS_VARS = {
+    'Binlog_cache_disk_use': ('mysql.binlog.cache_disk_use', GAUGE),
+    'Binlog_cache_use': ('mysql.binlog.cache_use', GAUGE),
+    'Handler_commit': ('mysql.performance.handler_commit', RATE),
+    'Handler_delete': ('mysql.performance.handler_delete', RATE),
+    'Handler_prepare': ('mysql.performance.handler_prepare', RATE),
+    'Handler_read_first': ('mysql.performance.handler_read_first', RATE),
+    'Handler_read_key': ('mysql.performance.handler_read_key', RATE),
+    'Handler_read_next': ('mysql.performance.handler_read_next', RATE),
+    'Handler_read_prev': ('mysql.performance.handler_read_prev', RATE),
+    'Handler_read_rnd': ('mysql.performance.handler_read_rnd', RATE),
+    'Handler_read_rnd_next': ('mysql.performance.handler_read_rnd_next', RATE),
+    'Handler_rollback': ('mysql.performance.handler_rollback', RATE),
+    'Handler_update': ('mysql.performance.handler_update', RATE),
+    'Handler_write': ('mysql.performance.handler_write', RATE),
+    'Opened_tables': ('mysql.performance.opened_tables', RATE),
+    'Qcache_total_blocks': ('mysql.performance.qcache_total_blocks', GAUGE),
+    'Qcache_free_blocks': ('mysql.performance.qcache_free_blocks', GAUGE),
+    'Qcache_free_memory': ('mysql.performance.qcache_free_memory', GAUGE),
+    'Qcache_not_cached': ('mysql.performance.qcache_not_cached', RATE),
+    'Qcache_queries_in_cache': ('mysql.performance.qcache_queries_in_cache', GAUGE),
+    'Select_full_join': ('mysql.performance.select_full_join', RATE),
+    'Select_full_range_join': ('mysql.performance.select_full_range_join', RATE),
+    'Select_range': ('mysql.performance.select_range', RATE),
+    'Select_range_check': ('mysql.performance.select_range_check', RATE),
+    'Select_scan': ('mysql.performance.select_scan', RATE),
+    'Sort_merge_passes': ('mysql.performance.sort_merge_passes', RATE),
+    'Sort_range': ('mysql.performance.sort_range', RATE),
+    'Sort_rows': ('mysql.performance.sort_rows', RATE),
+    'Sort_scan': ('mysql.performance.sort_scan', RATE),
+    'Table_locks_immediate': ('mysql.performance.table_locks_immediate', RATE),
+    'Threads_cached': ('mysql.performance.threads_cached', GAUGE),
+    'Threads_created': ('mysql.performance.threads_created', GAUGE)
```

Comment:
To monitor the number of threads in the cache and how many threads have been created because a thread could not be taken from the cache, monitor the `Threads_cached` and `Threads_created` status variables.

I believe `Threads_created` is not a rate, the value may fluctuate depending on how many connections we could attend from the thread pool. I think `Threads_cached` probably doesn't behave as a rate either.